qa/rpc-tests/test_framework/util.py

```diff
@@ -399,26 +399,29 @@ def wait_and_assert_operationid_status(node, myopid, in_status='success', in_err
     assert_true(result is not None, "timeout occured")
     status = result['status']
 
-    txid = None
+    ret = None
     errormsg = None
     if status == "failed":
         errormsg = result['error']['message']
     elif status == "success":
-        txid = result['result']['txid']
+        if type(result['result']) is dict and result['result'].get('txid'):
+            ret = result['result']['txid']
+        else:
+            ret = result['result']
```

Comment:
Which RPC required making this change?

Reply:
This change was to accommodate `AsyncRPCOperation_saplingmigration` which does not return a `txid` in it's result.

Reply:
This particular function is expected to return a `txid`, so if you want to access the entire result, you should refactor this into two functions: an inner function that returns the entire result, and the existing function that calls the inner function, then only returns the `txid` (and returns `None` in the case of `z_saplingmigration`).